web_timeline/static/src/js/timeline_renderer.js

```diff
@@ -151,6 +154,16 @@ var CalendarRenderer = AbstractRenderer.extend({
             onUpdate: self.on_update,
             onRemove: self.on_remove,
         });
+        if (this.arch.children.length) {
+            this.qweb = new QWeb(session.debug, {_s: session.origin}, false);
+            var tmpl = utils.json_node_to_xml(
+                _.filter(this.arch.children, function(item) {
+                    return item.tag === 'templates';
+                })[0]
+            );
+            this.qweb.add_template(tmpl);
```

Comment:
worth to validate the template names here?

Reply:
You can have multiple templates if you want, but timeline-item will be selected by the timeline. e.g. you can have:

```
<templates>
      <t t-name="timeline-item-progressbar">
           ...
      </t>
      <t t-name="timeline-item">
             <t t-call="timeline-item-progressbar"/>
      </t>
</templates>